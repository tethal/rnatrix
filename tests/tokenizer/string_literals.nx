// Basic string literals
"hello"
"world"
""

// Strings with escape sequences
"quote: \""
"backslash: \\"
"newline: \n"
"tab: \t"
"return: \r"
"null: \0"

// Multiple escapes in one string
"line1\nline2\nline3"
"path\\to\\file"
"\"quoted\""

// Complex combinations
"tab\tthen\"quote"
"\0\n\r\t\\\""