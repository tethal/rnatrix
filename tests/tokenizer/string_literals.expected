Token { tt: StringLiteral, span: @0:25-32, name: None }: "\"hello\""
Token { tt: StringLiteral, span: @0:33-40, name: None }: "\"world\""
Token { tt: StringLiteral, span: @0:41-43, name: None }: "\"\""
Token { tt: StringLiteral, span: @0:78-89, name: None }: "\"quote: \\\"\""
Token { tt: StringLiteral, span: @0:90-105, name: None }: "\"backslash: \\\\\""
Token { tt: StringLiteral, span: @0:106-119, name: None }: "\"newline: \\n\""
Token { tt: StringLiteral, span: @0:120-129, name: None }: "\"tab: \\t\""
Token { tt: StringLiteral, span: @0:130-142, name: None }: "\"return: \\r\""
Token { tt: StringLiteral, span: @0:143-153, name: None }: "\"null: \\0\""
Token { tt: StringLiteral, span: @0:189-210, name: None }: "\"line1\\nline2\\nline3\""
Token { tt: StringLiteral, span: @0:211-227, name: None }: "\"path\\\\to\\\\file\""
Token { tt: StringLiteral, span: @0:228-240, name: None }: "\"\\\"quoted\\\"\""
Token { tt: StringLiteral, span: @0:266-284, name: None }: "\"tab\\tthen\\\"quote\""
Token { tt: StringLiteral, span: @0:285-299, name: None }: "\"\\0\\n\\r\\t\\\\\\\"\""
Token { tt: Eof, span: @0:299-299, name: None }: ""
